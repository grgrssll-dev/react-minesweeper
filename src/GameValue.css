.menu-number {
	background: black;
	color: red;
	font-display: block;
    font-family: 'VT323', Monaco, monospace;
	font-size: 2rem;
	line-height: 0.7;
	height: 1.5rem;
	text-align: right;
	width: 2.5rem;
}