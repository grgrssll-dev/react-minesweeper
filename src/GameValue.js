import React, { PureComponent } from 'react';
import './GameValue.css';

import Utils from './Utils';

class GameValue extends PureComponent {

	render() {
		const { value } = this.props;
		return (
			<div className="menu-number">
				<span>{Utils.padLeft(value, 3, '0')}</span>
			</div>
		);
	}
}

export default GameValue;
