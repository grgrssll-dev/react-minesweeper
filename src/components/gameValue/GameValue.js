import React from 'react';
import './GameValue.css';

function gameValue(props) {
	const { value } = props;
	return (
		<div className="menu-number">
			<span>{`${value || ''}`.padStart(3, '0')}</span>
		</div>
	);
}

export default gameValue;
