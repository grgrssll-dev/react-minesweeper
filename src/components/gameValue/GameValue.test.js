import React from 'react';
import { shallow } from 'enzyme/build';
import GameValue from './GameValue';

const baseProps = {
	value: 123,
	className: 'class-name',
};

describe('GameValue [Unit]', () => {
	beforeEach(() => {
		jest.clearAllMocks();
	});

	it('should match component snapshot', () => {
		expect.assertions(1);
		const props = {
			...baseProps,
		};
		const wrapper = shallow(<GameValue {...props} />);
		expect(wrapper).toMatchSnapshot();
	});
});