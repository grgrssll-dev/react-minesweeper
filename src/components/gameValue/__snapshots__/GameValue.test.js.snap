// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameValue [Unit] should match component snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
